server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -358,7 +427,8 @@ static class WeightFunction {
             this.indexBalance = indexBalance;
             this.shardBalance = shardBalance;
             this.constraints = new AllocationConstraints();
-            this.rebalanceConstraints = new RebalanceConstraints();
+            this.rebalanceConstraints = new RebalanceConstraints(preferPrimaryBalanceBuffer);
+            this.preferPrimaryBalanceBuffer = preferPrimaryBalanceBuffer;
```

Comment:
if the primary rebalance setting is disabled, how are we making sure that the buffer setting does not work?

Reply:
The buffer won't be functional if the setting is not enabled:
```
    private void setPreferPrimaryShardRebalance(boolean preferPrimaryShardRebalance) {
        this.preferPrimaryShardRebalance = preferPrimaryShardRebalance;
        this.weightFunction.updateRebalanceConstraint(CLUSTER_PRIMARY_SHARD_REBALANCE_CONSTRAINT_ID, preferPrimaryShardRebalance);
    }
```